processor/src/test/java/org/mapstruct/ap/testutil/runner/GeneratedSource.java

```diff
@@ -86,4 +133,38 @@ public JavaFileAssert forDecoratedMapper(Class<?> mapperClass) {
     public JavaFileAssert forJavaFile(String path) {
         return new JavaFileAssert( new File( compilingStatement.get().getSourceOutputDir() + "/" + path ) );
     }
+
+    private class GeneratedSourceStatement extends Statement {
+        private final Statement next;
+
+        private GeneratedSourceStatement(Statement next) {
+            this.next = next;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+            next.evaluate();
+            handleFixtureComparison();
+        }
+    }
+
+    private void handleFixtureComparison() throws UnsupportedEncodingException {
+        for ( Class<?> fixture : fixturesFor ) {
+            String expectedFixture = getMapperName( fixture );
+            URL expectedFile = getClass().getClassLoader().getResource( expectedFixture );
```

Comment:
How about we put the fixture files in `src/test/resources/fixtures` or so, to clarify the purpose of the .java files that we're putting there?